Variant 13 Drujinin Igor

Drujininii opened the pull request:
Не уверен, что я туда все пушу. Но очень на это надеюсь...

avtyul commented on source/sortString.js:
```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const sortString = (str) => {
```
при одном аргументе скобки можно опустить

avtyul commented on source/sortString.js:
```diff
@@ -0,0 +1,13 @@
+
+const sortString = (str) => {
+  let words = str.split(' ');
+  for (let i in words) {
```
words - массив, у массива есть методы обхода. [некоторые даже возвращают новый массив](https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

avtyul commented on source/sortString.js:
```diff
@@ -0,0 +1,13 @@
+};
+
+const stringCompare = (lhs, rhs) => {
+  return lhs.localeCompare(rhs);
```
если в теле функции одна строка, можно сократить до `const foo = x => x`

avtyul commented on source/sortString.js:
```diff
@@ -0,0 +1,13 @@
+
+const stringCompare = (lhs, rhs) => {
+  return lhs.localeCompare(rhs);
+}
```
git ждет newline

avtyul commented on tests/sortString.js:
```diff
@@ -21,4 +21,9 @@ QUnit.module('Тестируем функцию sortString', function () {
 	});
+
+  QUnit.test('Функция работает верно', function (assert) {
+    assert.strictEqual(sortString('ママソープフレーム'), 'ーーソフプママムレ', 'Работает с японским алфавитом');
```
тебе все равно никто не поверит

avtyul reviewed (commented):
Привет, оставил замечания и не только ) Жду фиксы

avtyul commented on source/sortString.js:
```diff
@@ -0,0 +1,11 @@
+  	return word.split('').sort(stringCompare).join('');
+  });
+  return sortedWords.sort(stringCompare).join(' ');
+};
```
```javascript
const stringCompare = (lhs, rhs) => lhs.localeCompare(rhs);
const sortString = str => str.split(' ').map(word => word.split('').sort(stringCompare).join('')).sort(stringCompare).join(' ');
```

avtyul commented on tests/sortString.js:
```diff
@@ -21,4 +21,10 @@ QUnit.module('Тестируем функцию sortString', function () {
 	});
+
+	QUnit.test('Функция работает верно', function (assert) {
+  	assert.strictEqual(sortString('купи кипу пик'), 'икп икпу икпу', 'Язык ломается и там, и там, всё верно');
```
И еще раз респект за креативный тест

nksoff commented:
Твой код действительно работает, и наши тесты пройдены. Но в условии написано, что необходимо написать *парочку дополнительных тестов*.

nksoff commented on source/get.js:
```diff
@@ -0,0 +1,13 @@
+function get(obj, key_to_find) {
+	var arrProperty = key_to_find.split('.');//массив свойств
```
используй [let, const](https://learn.javascript.ru/let-const) вместо var

nksoff commented on source/get.js:
```diff
@@ -0,0 +1,13 @@
+	propertyObj = obj;//свойства объектов
+	if (key_to_find != '.') {
+		for (i = 1; i < arrProperty.length; i++) {//первый символ в массиве пустой, т.к. выражение начинается с точки
+			if (propertyObj == undefined) 
```
лучше использовать строгое сравнение: `===`. сравни: [==](https://i.stack.imgur.com/35MpY.png) vs [===](https://i.stack.imgur.com/62vxI.png)

nksoff commented on source/get.js:
```diff
@@ -0,0 +1,13 @@
+		}
+		return propertyObj;
+	};
+	return obj;
```
и вроде хорошо написано, но не хватает современного js :-)
будет здорово заменить условие `key_to_find != '.'` на фильтрацию непустых элементов массива `arrProperty`. тогда и цикл можно будет начать с `i = 0`.

Drujininii commented:
Когда дедлайн по этой задаче? Люблю я эти задания с дописанием тестов) думал, этого никто не заметит...тесты могут быть бессмысленными и беспощадными? Их добавить в имеющиеся категории?С уважением, Дружинин Игорь+7 985 449 66 4923:21, 31 августа 2017 г., Nikita Slimov <notifications@github.com>:@nksoff commented on this pull request.In source/get.js:> @@ -0,0 +1,13 @@+function get(obj, key_to_find) {+	var arrProperty = key_to_find.split('.');//массив свойствиспользуй let, const вместо varIn source/get.js:> @@ -0,0 +1,13 @@+function get(obj, key_to_find) {+	var arrProperty = key_to_find.split('.');//массив свойств+	propertyObj = obj;//свойства объектов+	if (key_to_find != '.') {+		for (i = 1; i < arrProperty.length; i++) {//первый символ в массиве пустой, т.к. выражение начинается с точки+			if (propertyObj == undefined) лучше использовать строгое сравнение: ===. сравни: == vs ===In source/get.js:> @@ -0,0 +1,13 @@+function get(obj, key_to_find) {+	var arrProperty = key_to_find.split('.');//массив свойств+	propertyObj = obj;//свойства объектов+	if (key_to_find != '.') {+		for (i = 1; i < arrProperty.length; i++) {//первый символ в массиве пустой, т.к. выражение начинается с точки+			if (propertyObj == undefined) +				break;+			propertyObj = propertyObj[arrProperty[i]];+		}+		return propertyObj;+	};+	return obj;и вроде хорошо написано, но не хватает современного js :-)будет здорово заменить условие key_to_find != '.' на фильтрацию непустых элементов массива arrProperty. тогда и цикл можно будет начать с i = 0.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

nksoff commented:
Ааа. Как ужасно парсятся письма!

Тесты должны быть не совсем бесполезные. Покрой те входные данные, которые еще не покрыты.

Drujininii commented:
Добавил два теста. Они не особо умны, конечно. Сделал скрипт помоднее. Вроде отправил пулл реквест. Он пишет, что делает пулл реквест автоматически, поэтому не всегда уверен в правильности операции-- 01.09.2017, 10:03, "Nikita Slimov" <notifications@github.com>:Ааа. Как ужасно парсятся письма!
Тесты должны быть не совсем бесполезные. Покрой те входные данные, которые еще не покрыты.

—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

nksoff commented on source/get.js:
```diff
@@ -1,13 +1,11 @@
-	};
-	return obj;
-}
+	for (i = 0; i < arrProperty.length; i++) {//первый символ в массиве пустой, т.к. выражение начинается с точки
```
комментарий соответствует тому, что происходит?

Drujininii commented:
Забыл...ща подправим) мой косяк)С уважением, Дружинин Игорь+7 985 449 66 4911:51, 1 сентября 2017 г., Nikita Slimov <notifications@github.com>:@nksoff commented on this pull request.In source/get.js:>  	propertyObj = obj;//свойства объектов-	if (key_to_find != '.') {-		for (i = 1; i < arrProperty.length; i++) {//первый символ в массиве пустой, т.к. выражение начинается с точки-			if (propertyObj == undefined) -				break;-			propertyObj = propertyObj[arrProperty[i]];-		}-		return propertyObj;-	};-	return obj;-}+	for (i = 0; i < arrProperty.length; i++) {//первый символ в массиве пустой, т.к. выражение начинается с точкикомментарий соответствует тому, что происходит?—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

nksoff commented on source/get.js:
```diff
@@ -2,7 +2,7 @@ function get(obj, key_to_find) {
 	arrProperty = arrProperty.filter(property => property != '');
 	propertyObj = obj;//свойства объектов
-	for (i = 0; i < arrProperty.length; i++) {
+	for (i = 0; i < arrProperty.length; i++) {//в цикле спускаемся по вложенным свойствам
```
комментарий бесполезен.

а еще в этой строчке кроется проблемка: перед `i = 0` нет ни `var`, ни `let`, ни `const`. ты используешь глобальную переменную `i`.

Примерно так можно это увидеть:
```js
const f = () => {
    for (i = 0; i < 10; i++) {
        console.log('inner: i=', i);
    }
};

f();
console.log('outer: i=', i); // outer: i= 10
```

nksoff commented:
# Ок
